Cargo.toml

```diff
@@ -66,6 +66,7 @@ packed_simd = { version = "0.3.4", package = "packed_simd_2", features = ["into_
 default = ["alloc", "precomputed-tables", "zeroize"]
 alloc = ["zeroize?/alloc"]
 precomputed-tables = []
+legacy_compatibility = []
```

Comment:
I don't love the naming of this. It matches the `ed25519-dalek` one though. Thoughts?

Reply:
Matching the name in `ed25519-dalek` seems fine.

I was thinking it might also be good to explicitly deprecate the methods that are feature-gated by this too.

Reply:
Wait, do we intend to remove this function? Are we deprecating `ed25519-dalek` legacy compat soon?

Reply:
My worry is more feature flags are bad at gating APIs we don't want people to use because they're additive, so one crate in the project activating the feature makes the functionality available to all.

Deprecation is a bit clearer about that.

Reply:
Isn't it noisy though? Like if you pull in a library that has to accept legacy signatures, you're going to see a compiler warning every time

Reply:
We can continue the conversation here. I'm gonna merge the rest of it for now so I can update the upstream stuff

Reply:
Warnings aren't shown in dependencies by default in Rust+Cargo (for this reason).

The client can use `#[allow(deprecated)]` too (and probably will).

Reply:
Ah, nice! Ok adding now.